data-prepper-plugins/kafka-connect-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafkaconnect/source/KafkaConnectSourceTest.java

```diff
@@ -0,0 +1,184 @@
+public class KafkaConnectSourceTest {
+    private final String TEST_PIPELINE_NAME = "test_pipeline";
+    private KafkaConnectSource kafkaConnectSource;
+
+    @Mock
+    private MySQLConfig mySQLConfig;
+
+    @Mock
+    private KafkaConnectConfig kafkaConnectConfig;
+
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private PipelineDescription pipelineDescription;
+
+    @Mock
+    private Buffer<Record<Object>> buffer;
+
+    @Mock
+    private KafkaConnect kafkaConnect;
+
+    @Mock
+    private KafkaClusterConfigSupplier kafkaClusterConfigSupplier;
+
+    @Mock
+    private KafkaConnectConfigSupplier kafkaConnectConfigSupplier;
+
+    private String bootstrapServers = "localhost:9092";
+
+    public KafkaConnectSource createSourceUnderTest() {
+        return new MySQLSource(mySQLConfig, pluginMetrics, pipelineDescription, kafkaClusterConfigSupplier, kafkaConnectConfigSupplier);
+    }
+
+    @BeforeEach
+    void setUp() {
+        WorkerProperties workerProperties = new WorkerProperties();
+        workerProperties.setBootstrapServers(bootstrapServers);
+        kafkaConnectConfigSupplier = mock(KafkaConnectConfigSupplier.class);
+        lenient().when(kafkaConnectConfigSupplier.getConfig()).thenReturn(kafkaConnectConfig);
+        lenient().when(kafkaConnectConfig.getWorkerProperties()).thenReturn(workerProperties);
+        lenient().when(mySQLConfig.buildConnectors()).thenReturn(Collections.emptyList());
+
+        pipelineDescription = mock(PipelineDescription.class);
+        lenient().when(pipelineDescription.getPipelineName()).thenReturn(TEST_PIPELINE_NAME);
+        pluginMetrics = mock(PluginMetrics.class);
+    }
+
+    @Test
+    void testStartKafkaConnectSource() {
+        try (MockedStatic<KafkaConnect> mockedStatic = mockStatic(KafkaConnect.class);
+             MockedStatic<KafkaSourceSecurityConfigurer> mockedSecurityConfigurer = mockStatic(KafkaSourceSecurityConfigurer.class)) {
+            mockedSecurityConfigurer.when(() -> KafkaSourceSecurityConfigurer.setAuthProperties(any(), any(), any())).thenAnswer((Answer<Void>) invocation -> null);
+            kafkaConnect = mock(KafkaConnect.class);
+            doNothing().when(kafkaConnect).addConnectors(any());
+            doNothing().when(kafkaConnect).start();
+            doNothing().when(kafkaConnect).stop();
+            // Set up the mock behavior for the static method getInstance()
+            mockedStatic.when(() -> KafkaConnect.getPipelineInstance(any(), any(), any(), any())).thenReturn(kafkaConnect);
+            kafkaConnectSource = createSourceUnderTest();
+            kafkaConnectSource.start(buffer);
+            verify(kafkaConnect).addConnectors(any());
+            verify(kafkaConnect).start();
+            try {
+                Thread.sleep(10);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            kafkaConnectSource.stop();
+            verify(kafkaConnect).start();
```

Comment:
this should be `verify(kafkaConnect).stop() `?

Reply:
Yep, ty